### PR TITLE
fix(tests): Improve header testing

### PR DESCRIPTION
**Pull request by nick-w-nick**
I noticed that the header tests could theoretically be improved/refactored by moving the input/output logic into an array and then generating the tests using `it.each()`.

I also added an empty value `""` to the array test to ensure empty header value logic is properly tested for arrays.

### CODE REVIEW

File: test/index.test.ts [hunk 1]
```diff
@@ -31,41 +31,44 @@ describe("fetchNodeRequestHandler", () => {
...
-      headers: { foo: "bar", host: "localhost" },
-    });
-  });
-
-  it("headers (array)", async () => {
-    const res = await fetchNodeRequestHandler(echoHandler, "/test", {
-      headers: [
+  const requestHeaderTestCases: {
+    description: string;
+    input: Record<string, string> | Headers | [string, string][];
```
[the remark this answers is not in the document]
Comment (nick-w-nick) [reply test/index.test.ts]: @pi0 Updated, thanks for the heads up!

File: test/index.test.ts [hunk 2]
```diff
@@ -31,41 +31,44 @@ describe("fetchNodeRequestHandler", () => {
...
-    expect(await res.json()).toEqual({
-      url: "/test",
-      headers: { foo: "bar", host: "localhost", array: ["a", "b", "c"] },
-    });
-  });
+      expected: { foo: "bar", host: "localhost", array: ["a", "b", "c"] },
+    },
+  ];
+
+  it.each(requestHeaderTestCases)(
```
[the remark this answers is not in the document]
Comment (nick-w-nick) [reply test/index.test.ts]: Great suggestion, you got it!

I've updated the loop to be a traditional `for...of` loop rather than a vitest `it.each` loop.

[the remark this answers is not in the document]
Comment (nick-w-nick) [reply test/index.test.ts]: @pi0 Great suggestion, you got it!

I've updated the loop to be a traditional `for...of` loop rather than a vitest `it.each` loop.

**Review by pi0 — APPROVED** [review]
Thanks!
